src/registries/figure_registry.ts

```diff
@@ -15,17 +19,119 @@ import { Registry } from "./registry";
 
 export interface FigureContent {
   Component: any;
+  menuRegistryBuilder: (
+    figureId: UID,
+    onFigureDeleted: () => void,
+    env: SpreadsheetChildEnv
+  ) => MenuItemRegistry;
```

Comment:
do we we need a registry ?
```suggestion
  menuBuilder: (
    figureId: UID,
    onFigureDeleted: () => void,
    env: SpreadsheetChildEnv
  ) => MenuItem[];
```

Reply:
No :)